### PR TITLE
making better use of package.json

### PR DESCRIPTION
**Pull request by fearphage**
- removed unused dependencies (`eslint-plugin-react`)
- upgraded to `babel-register` from `babel-core/register`
  - switch to using npm event hooks instead of multiple commands strung together

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,10 @@
   "scripts": {
     "compile": "babel -d dist src",
     "lint": "eslint .",
-    "test": "npm run lint && mocha --compilers js:babel-core/register --recursive",
-    "test:ci": "npm run compile && npm test"
+    "pretest": "npm run lint",
```
Comment (Chris911) [inline package.json:9]: Is `pretest` officially supported for node modules?

Comment (fearphage) [inline package.json:9]: Yes, it is. https://docs.npmjs.com/misc/scripts

File: package.json [hunk 2]
```diff
@@ -29,12 +31,11 @@
   },
   "devDependencies": {
     "babel-cli": "^6.6.5",
-    "babel-core": "^6.7.2",
     "babel-preset-es2015": "^6.6.0",
+    "babel-register": "^6.7.2",
     "chai": "^3.5.0",
     "eslint": "2.3.0",
     "eslint-config-airbnb": "6.1.0",
-    "eslint-plugin-react": "^4.2.3",
```
Comment (Chris911) [inline package.json:37]: Everything seems to work but I'm getting the following warning when installing with node 5 and npm 3:
screen shot 2016-04-03 at 1 35 01 pm

Comment (fearphage) [inline package.json:37]: It's unnecessary when using [AirBnB's base config](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb#eslint-config-airbnbbase).

**Comment by Chris911** [comment]
Thanks :+1:

**Pull request merged**